Reference/google/protobuf/unittest_custom_options.pb.swift

```diff
@@ -1684,14 +1906,24 @@ public struct ProtobufUnittest_OldOptionType: ProtobufGeneratedMessage {
 
   }
 
-  public var value: ProtobufUnittest_OldOptionType.TestEnum = ProtobufUnittest_OldOptionType.TestEnum.oldValue
+  private var _value: ProtobufUnittest_OldOptionType.TestEnum? = nil
+  public var value: ProtobufUnittest_OldOptionType.TestEnum {
+    get {return _value ?? ProtobufUnittest_OldOptionType.TestEnum.oldValue}
+    set {_value = newValue}
+  }
+  public var hasValue: Bool {
+    return _value != nil
+  }
+  public mutating func clearValue() {
+    return _value = nil
+  }
 
   public init() {}
 
   public mutating func _protoc_generated_decodeField(setter: inout ProtobufFieldDecoder, protoFieldNumber: Int) throws -> Bool {
     let handled: Bool
     switch protoFieldNumber {
-    case 1: handled = try setter.decodeRequiredField(fieldType: ProtobufUnittest_OldOptionType.TestEnum.self, value: &value)
+    case 1: handled = try setter.decodeOptionalField(fieldType: ProtobufUnittest_OldOptionType.TestEnum.self, value: &_value)
```

Comment:
Decoding of individual fields doesn't need to distinguish between the optional-/required-ness of the field; this should be done with an `isInitialized` check at the end of decoding (and only then in the non-partial parsing case).

I could have made this clearer in the original PR comment; one side effect of changing the optionality of the properties in this PR was that `decodeRequiredField` is no longer required and I could use `decodeOptionalField` everywhere. After this change we should delete `decodeRequiredField` and rename `decodeOptionalField` to `decodeSingularField`, but I wanted to keep this PR as targeted as possible.